src/components/Store.test.js

```diff
@@ -0,0 +1,52 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Store from "./Store";
+
+describe("<Store />", () => {
+  beforeEach(() => {
+    render(<Store />);
```

Comment:
Em vez de um beforeEach, o que acha de definirmos uma função daquela de setup para que a gente controle _quando_ durante o teste o elemento vai ser realmente renderizado? Existe uma boa prática de teste chamada AAA (arrange-act-assert). Que define que devemos fazer nesta ordem:

- arrange: a preparação dos dados que vamos usar para executar o nosso teste (isso pode ficar num before)
- act: a execução do que vamos realmente testar, por exemplo renderizar um componente e interagir com  ele  (isso tem que acontecer dentro do `it`)
- assert: checar as condições de sucesso (também dentro do `it`)

Aqui tem um pouco mais sobre o assunto https://medium.com/@pablodarde/o-padr%C3%A3o-triple-a-arrange-act-assert-741e2a94cf88. Os exemplos não estão em JS mas eles valem para qualquer linguagem.